ops/terraform/services/server/modules/bfd_server_asg/main.tf

```diff
@@ -230,65 +259,71 @@ resource "aws_cloudwatch_metric_alarm" "filtered_networkin_low" {
       }
       metric_name = "NetworkOut"
       namespace   = "AWS/EC2"
-      period      = 60
+      period      = local.scaling_alarms_config.scale_in.eval_period
+      stat        = "Average"
+    }
+  }
+
+  metric_query {
+    id          = "m3"
+    return_data = false
+
+    metric {
+      dimensions = {
+        AutoScalingGroupName = aws_autoscaling_group.main.name
+      }
+      metric_name = "GroupDesiredCapacity"
+      namespace   = "AWS/AutoScaling"
+      period      = local.scaling_alarms_config.scale_in.eval_period
       stat        = "Average"
     }
   }
 
   metric_query {
     expression  = "IF(m2/m1 > 0.01, m1, 0)"
-    id          = "e1"
+    id          = "networkin"
     label       = "FilteredNetworkIn"
+    return_data = false
+  }
+
+  metric_query {
+    id    = "e2"
+    label = "Set to ${each.value.desired_capacity} capacity units"
+    expression = "IF(${join(" && ", compact([
+      each.value.begin > 0 ? "networkin > ${each.value.begin}" : null,
```

Comment:
Not sure if it's even possible, but if it is, should we have `begin` be inclusive and `end` exclusive? Just in case `networkin` is ever 0?

Reply:
It's definitely possible, but would require the corresponding scale-out stages be swapped as well. If the ranges don't match between the two, we end up with inconsistent behavior if `NetworkIn` is ever _exactly_ on the boundary of a given range (i.e. 100 MB/min, 200 MB/min, etc.).

For this particular case, the ternary ensures that if the alarm's stage has a range with a `begin` of 0, the generated Metric Math looks like this:

```
IF(networkin <= {END} && m3 > {DESIRED_CAPACITY}, 1)
```
so that _any_ `networkin` beneath `end` will return data. Obviously, this is different from:
```
IF(networkin > 0 && networkin <= {END} && m3 > {DESIRED_CAPACITY}, 1)
```
which would result in a `networkin` of 0 returning no data (thus "breaking" consecutive periods of evaluation and forcing the alarm to start re-evaluating from the start of a new 15 minute period).

We should be covered by the above, but I have no problem inverting the range inclusiveness to make `begin` inclusive and `end` exclusive if that would be more understandable. We would be able to remove this ternary, for example.

Reply:
I think all is well with what you have here :) No need to go and change things up when they're perfectly functional as-is